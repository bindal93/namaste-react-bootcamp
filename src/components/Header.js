import { title } from "../utils/constants.js";
import { Link } from "react-router-dom";

const HeadingComponent = () => (
  <div id="title" className="title-class" tabIndex="1">
    <Link to="/">
      <h2>{title}</h2>
    </Link>
    <Link to="/search">
      <span>Search</span>
    </Link>
    <Link to="/about-us">
      <span>AboutUs</span>
    </Link>
  </div>
);

export default HeadingComponent;
