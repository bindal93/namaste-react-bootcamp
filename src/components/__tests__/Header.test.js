import Header from "../Header";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../store";
import { StaticRouter } from "react-router-dom/server";

const logoUrl =
  "https://uploads-ssl.webflow.com/6080464681216414c7d1d60b/6080468fecb6a579c58f35f4_hungerhub%20logo.png";

test("Verify if Header Loads", async () => {
  const header = render(
    <StaticRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </StaticRouter>
  );

  const logo = await header.findByTestId("logo");

  expect(logo.src).toBe(logoUrl);
});

test("Verify if Header Loads", async () => {
  const header = render(
    <StaticRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </StaticRouter>
  );

  expect(header).toMatchSnapshot();
});
