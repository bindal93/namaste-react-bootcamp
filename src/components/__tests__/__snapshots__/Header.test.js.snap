// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify if Header Loads 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex px-10 py-4 justify-between bg-purple-50"
        id="title"
        tabindex="1"
      >
        <a
          href="/"
        >
          <img
            class="h-12"
            data-testid="logo"
            src="https://uploads-ssl.webflow.com/6080464681216414c7d1d60b/6080468fecb6a579c58f35f4_hungerhub%20logo.png"
          />
        </a>
        <div
          class="flex space-x-8 cursor-pointer"
        >
          <a
            href="/search"
          >
            <span>
              Search
            </span>
          </a>
          <a
            data-testid="about"
            href="/about-us"
          >
            <span>
              AboutUs
            </span>
          </a>
          <a
            href="/search"
          >
            <span>
              Contact Us
            </span>
          </a>
          <a
            href="/cart"
          >
            <span>
              Cart - 
              1
               items
            </span>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex px-10 py-4 justify-between bg-purple-50"
      id="title"
      tabindex="1"
    >
      <a
        href="/"
      >
        <img
          class="h-12"
          data-testid="logo"
          src="https://uploads-ssl.webflow.com/6080464681216414c7d1d60b/6080468fecb6a579c58f35f4_hungerhub%20logo.png"
        />
      </a>
      <div
        class="flex space-x-8 cursor-pointer"
      >
        <a
          href="/search"
        >
          <span>
            Search
          </span>
        </a>
        <a
          data-testid="about"
          href="/about-us"
        >
          <span>
            AboutUs
          </span>
        </a>
        <a
          href="/search"
        >
          <span>
            Contact Us
          </span>
        </a>
        <a
          href="/cart"
        >
          <span>
            Cart - 
            1
             items
          </span>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
